src/wallet/rpcdump.cpp

```diff
@@ -197,8 +201,12 @@ UniValue importaddress(const UniValue& params, bool fHelp)
             throw JSONRPCError(RPC_WALLET_ERROR, "The wallet already contains the private key for this address or script");
 
         // add to address book or update label
-        if (address.IsValid())
-            pwalletMain->SetAddressBook(address.Get(), strLabel, AddressBook::AddressBookPurpose::RECEIVE);
+        if (address.IsValid()) {
+            std::string strPurpose = AddressBook::AddressBookPurpose::RECEIVE;
+            if (address.IsStakingAddress())
+                strPurpose = AddressBook::AddressBookPurpose::COLD_STAKING;
```

Comment:
Minor nit, inline this strPurpose initialization.

Reply:
or (minor nit) use same code style as DecodeDumpString()
```
        if (address.IsValid()) {
                pwalletMain->SetAddressBook(address.Get(), strLabel, address.IsStakingAddress() ?
                        AddressBook::AddressBookPurpose::COLD_STAKING :
                        AddressBook::AddressBookPurpose::RECEIVE));
            }